algorithms/rainbow/agent.py

```diff
@@ -0,0 +1,158 @@
+"""
+Adapted from https://github.com/Kaixhin/Rainbow
+
+Wrapper class including setup, training and evaluation functions for Rainbow DQN model
+"""
+import os
+import numpy as np
+import torch
+from torch import optim
+
+from algorithms.rainbow.model import RainbowDQN
+
+
+class Agent:
+    """
+    Wraps control between both online and target network for setup, training and evaluation
+    """
+    def __init__(self, args, env):
+        """
+        Q(s,a) is the expect reward. Z is the full distribution from which Q is generated
+        Support represents the support of Z distribution
```

Comment:
from the distributional blog:

"If p and q are two distributions with same support (i.e. their pdfs are non-zero at the same points)"
the part in brackets seems like something we can throw in

Reply:
Isn't this going too far? explaining what the support of a random variable is